packages/block-library/src/image/edit.js

```diff
@@ -275,6 +276,7 @@ export function ImageEdit( {
 	);
 
 	const classes = classnames( className, {
+		[ duotone?.id ]: duotone,
```

Comment:
In #29335 I need a similar unique id to apply inline style for the current block, Do you think we can build a generic thing for that and maybe extract it to its own PR first. It could be a support flag on its own or something.

Reply:
The IDs here don't really need to be completely unique—they just need to be unique to the color combination.

I like the readability of the generated styles this way rather than a pseudo-random unique ID. When I demoed the early prototype internally at Automattic, I was using the block ID as a unique ID for the stylesheet, but I received feedback that it would be nice if the duplicate duotones could be combined into a single SVG and stylesheet for the page.

If we were doing something generic for outside usage, I think it might be better if we could just have some sort of hashing function rather than generating UUIDs so that we might still be able to do that deduplication eventually. Although, I'm still not really sure how that deduplication would work.

Reply:
I personally think unique ids are better because conceptually this is an inline style specific to the current block.
The SVG is not though and yes, it would be good if we can dedup the SVGs (they are duplicated even now) but I don't think it makes sense to base the classname added to the block on the value because we'll be adding a classname per inline style while it's better to have a unique classname for the block and just apply on  the inline styles to the same classname.

Reply:
> conceptually this is an inline style specific to the current block

This is very styling-centric, so conceptually for me, this is the same as using something like `hue-rotate` in CSS. If I was writing this HTML and CSS from scratch, I'd probably have a classname to apply the style, especially if I only have a couple different filters that I would want to apply to multiple images.

It also reminds me that it's possible to [inline the style within the CSS](https://css-tricks.com/svg-within-css/). If it were implemented that way, would you think differently about it?

Reply:
> It also reminds me that it's possible to inline the style within the CSS. If it were implemented that way, would you think differently about it?

Potentially yes, but only for presets, for custom duotones, It seems that it's still better to me to have a unique class for the block. Basically, I see it as the same thing as an "inline style" applied to the element `style=""` since it's a custom value, the only difference is that we can't really use that because we need to target "sub items".

Reply:
Thinking more here, I can see the point though about considering these as actually "presets" but they are just presets generated dynamically in a single  page/post. The problem with that is that right now, they are being generated in the block itself so they are block specific where they shouldn't be if we follow that principle.

So it's one of two things for me:

 - Embrace the fact that these are inline styles and use a unique id.
 - Consider these as "dynamic presets" and in that case don't inline the style but instead have a way to "register dynamic custom presets" which means we can dedupe as well.

I still lean towards 1 because it's how we approached things so far for block customizations, and I don't want us to introduce an API/behavior without thinking holistically about it and its impact.

Reply:
> I see it as the same thing as an "inline style" applied to the element

I think I follow your reasoning now. It's more like an inline style because if you change it for one block, it's not going to affect the rest of the blocks that use the same color combination. A thing like "dynamic presets" would probably have to allow for updating everywhere when you update the preset, for example. Centralizing those things as "dynamic presets" would probably make deduplication easier too.

> Do you think we can build a generic thing for that and maybe extract it to its own PR first
 
I don't know if it really needs it. Is it good enough to just use the block id or `uuid()` like you have in #29335 instead? That's already pretty simple.

Reply:
yes, I think we should start with just a uuid at first, (left some related comments on the block supports PR)